components/Deck/ContentModulesPanel/CollectionsPanel/CollectionsPanel.js

```diff
@@ -0,0 +1,171 @@
+        let collectionDetails = {};
+
+        // transform collection details from array to json
+        collectionDetails = this.userCollections.reduce((details, value, key) => {
+            details[value._id] = value; return details;
+        }, {});
+
+        let collectionToAdd = collectionDetails[collectionId];
+        this.context.executeAction(addDeckToCollection, {
+            collectionId: collectionToAdd._id,
+            deckId: deckId,
+            collection: collectionToAdd,
+        });
+
+        this.setState({
+            currentSelection: ''
+        });
+    }
+    render() {
+        if (this.props.DeckCollectionStore.removeDeckFromCollectionError) {
+            this.showErrorPopup(this.context.intl.formatMessage(this.messages.removeDeckError));
+        } else if (this.props.DeckCollectionStore.addDeckToCollectionError) {
+            this.showErrorPopup(this.context.intl.formatMessage(this.messages.addDeckError));  
+        }
+
+        const userId = this.props.UserProfileStore.userid;
+        const selector = this.props.DeckCollectionStore.selector;
+        const groups = this.props.UserProfileStore.user.groups;
+        const groupIds = (groups || []).map( (group) => group.id);
+
+        // collections of the current deck
+        const deckCollections = this.props.DeckCollectionStore.deckCollections;
+
+        // collections that the user has edit rights
+        this.userCollections = [];
+        if (this.props.DeckCollectionStore.collections !== undefined) {
+            this.userCollections = this.props.DeckCollectionStore.collections.documents;
+        }
+        
+        // find collection dropdown options
+        let deckCollectionIds = deckCollections.map( (col) => col._id);
+        let collectionDropdownOptions = this.userCollections.filter( (collection) => {
+
+            // exclude collections that are already selected
+            return !deckCollectionIds.includes(collection._id);
+        }).map( (collection) => {
+            return {
+                key: collection._id,
+                value: collection._id,
+                text: collection.title
+            };
+        });
+
+        return (
+            <div className="ui bottom attached" ref="tagsPanel">
+                <div className="ui stackable grid">
+                    <div className="row">
+                        <div className="middle aligned eight wide column">
+                            <h3 className="ui header">{this.context.intl.formatMessage(this.messages.header)}</h3>
+                        </div>
+                    </div>
+                    {   (userId) &&
+                        <div className="row">
+                            <div className="eleven wide column">
+                                <Dropdown value={this.state.currentSelection} placeholder='Select to add a playlist' fluid search selection options={collectionDropdownOptions} onChange={this.addCollection.bind(this, selector.sid)} />
```

Comment:
Making the onChange here an onClick or onEnter might solve the keyboard selection problem

Reply:
I managed to solve that by using the semantic ui dropdown and not the react-semantic-ui one. Please check again